include/alpaka/kernel/KernelBundle.hpp

```diff
@@ -30,9 +63,21 @@ namespace alpaka
             : m_kernelFn(std::move(kernelFn))
             , m_args(std::forward<TArgs>(args)...)
         {
+#if BOOST_COMP_NVCC
```

Comment:
The asserts here are fine but why due you sneaked these into this PR? They are ortogonal to all other changes here and we need to remove the ckecks for cuda where we did it in the past.